modifiers/hwloc/modifier.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+from ramble.modkit import *
+
+
+class Hwloc(BasicModifier):
+    """Define a modifier for showing the underlying infrastructure topology"""
+
+    name = "hwloc"
+
+    mode(
+        name="on",
+        description="Mode for executing hwloc command",
+    )
+
+    executable_modifier("hwloc")
+
+    def hwloc(self, executable_name, executable, app_inst=None):
+        import os
+        from ramble.util.executable import CommandExecutable
+
+        hwloc_parser_dir = os.path.dirname(f"{self._file_path}")
+        hwloc_output_xml_file = f"{{experiment_run_dir}}/hwloc.{self._usage_mode}.xml"
+        hwloc_output_json_file = Path(hwloc_output_xml_file).with_suffix(".json")
+
+        pre_exec = []
+        post_exec = []
+
+        pre_exec.append(
+            CommandExecutable(
+                name="record_start_time", template=["start_time=$(date +%s%N)"]
+            )
+        )
+
+        # Run the hwloc tool and save its output in XML format to a file
+        pre_exec.append(
+            CommandExecutable(
+                "lstopo-to-get-underlying-infrastructure",
+                template=[
+                    f"lstopo --of xml --whole-system --whole-io --verbose {hwloc_output_xml_file} 2> /dev/null"
+                ],
+            )
+        )
+
+        caliper_modifier = any(
+            [modifier["name"] == "caliper" for modifier in app_inst.modifiers]
+        )
```

Comment:
This exact code snippet is dulicated in the `affinity modifier`
I tried adding a `modifiers/utils.py` and have all helpful common funcs our modifiers need in it. The probelm lies in ramble, as importing the utils.py to the modifiers didn't work as ramble has a different working directory path from ours. There were some hacks to solve this but were not clean nor safe at all.
I reached out to the ramble team here: [Ramble Issue](https://github.com/GoogleCloudPlatform/ramble/issues/1051) and will check their response. (I just gave this lower priority compared to other goals we are working on)